apps/backend/src/services/device_service.py

```
@@ -64,7 +61,7 @@ async def create_device(self, device_data: DeviceCreate) -> Device:
         except IntegrityError as e:
             await self.db.rollback()
             logger.error(f"Database integrity error creating device: {e}")
-            raise DatabaseOperationError(operation="create_device", details=str(e))
+            raise DatabaseOperationError(message="Failed to create device", operation="create_device", details={"error": str(e)})
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)